### PR TITLE
Initial Spec

### PR DESCRIPTION
**Pull request by ymalik**

### CODE REVIEW

File: scroll-anchoring/spec.html [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+  <head>
+    <title>Scroll Anchoring</title>
+    <meta charset='utf-8'>
+    <script src='https://www.w3.org/Tools/respec/respec-w3c-common'
+            async class='remove'></script>
+    <script class='remove'>
+      var respecConfig = {
+          specStatus: "CG-DRAFT",
+          shortName: "scrollAnchoring",
+          edDraftURI: "https://cdn.rawgit.com/ymalik/interventions/master/scroll-anchoring/spec.html",
```
Comment (marcoscaceres) [inline scroll-anchoring/spec.html]: Wonder if we should just use a gh-pages branch?

File: scroll-anchoring/spec.html [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        This spec also proposes an API for web developers to opt-out of this behavior.
+      </p>
+    </section>
+
+    <section id="intro" class="informative">
+      <h2>Introduction</h2>
+      <p>
+        Today, users of the web are often distracted by content moving around due to
+        changes that occur outside the viewport.  Examples include script inserting an
+        iframe containing an ad, or non-sized images loading on a slow network.
```
Comment (marcoscaceres) [inline scroll-anchoring/spec.html]: Nit: whitespace before "Examples".

Comment (marcoscaceres) [inline scroll-anchoring/spec.html]: oh, I see you use two spaces after a "." as a convention. Any reason why?

File: scroll-anchoring/spec.html [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+      </p>
+      <p>
+        Historically the browser's default behavior has been to preserve the absolute
+        scroll position when such changes occur.  This means that to avoid shifting
+        content, the webpage must take pains to reserve space on the page for anything
```
Comment (marcoscaceres) [inline scroll-anchoring/spec.html]: Avoid RFC2119 keywords (must) in non-normative prose.

File: scroll-anchoring/spec.html [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+        that will load later.  In practice, few websites do this consistently.
+      </p>
+      <p>
+        Scroll anchoring aims to minimize surprising content shifts by scrolling to
```
Comment (marcoscaceres) [inline scroll-anchoring/spec.html]: The paragraph is a bit hard to parse... it could be grammatically improved a little bit. 

Maybe something like:
>  Scroll anchoring aims to minimize surprising content shifts. It does this by anchoring ...

File: scroll-anchoring/spec.html [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+        follow the movement of visible page element.
+      </p>
+    </section>
+    <section>
+      <h2>Description</h2>
+      <section dfn-for="ScrollAnchoring">
+        <p>
+          Scroll anchoring works by selecting a DOM node (the <b>anchor node</b>) whose
```
Comment (marcoscaceres) [inline scroll-anchoring/spec.html]: Maybe use `<dfn>` instead of `<b>`, as you are defining what an anchor node is here.

File: scroll-anchoring/spec.html [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+          movement is used to determine the adjustment to the scroll position.
+        </p>
+        <h3>Anchor Node Selection</h3>
+        <p>
+          We aim to select an anchor node that is deep in the DOM and close to the top
```
Comment (marcoscaceres) [inline scroll-anchoring/spec.html]: Who is we? the UA?

File: scroll-anchoring/spec.html [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+          The current algorithm to select an anchor node for a scrollable area (either a
+          document or an element with scrollable overflow) is as follows:
+          <ol>
+            <li>Walk the DOM within the scrollable area.</li>
+            <li>
+              For each element E encountered by the walk
+              <ol>
+                <li>
+                  If E's style makes it an unsuitable anchor node, skip over E
+                  and its descendents.  E's style makes it an unsuitable anchor
```
Comment (marcoscaceres) [inline scroll-anchoring/spec.html]: I would recommend moving the "unsuitable anchor" test to its own sub section, as an algorithm. Then call into it by passing E.

File: scroll-anchoring/spec.html [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+        <h3>Anchor Node Selection</h3>
+        <p>
+          We aim to select an anchor node that is deep in the DOM and close to the top
+          edge of the viewport.  This maximizes the amount of offscreen content whose
+          changes we will successfully compensate for, while avoiding undesired scrolling
+          when important content loads within the viewport (for example, upon clicking a
+          "Read More" link).
+        </p>
+        <p>
+          The current algorithm to select an anchor node for a scrollable area (either a
```
Comment (marcoscaceres) [inline scroll-anchoring/spec.html]: Drop "current", add `<dfn>algorithm to select an anchor node for a scrollable area</dfn>`

File: scroll-anchoring/spec.html [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+            <li>Walk the DOM within the scrollable area.</li>
+            <li>
+              For each element E encountered by the walk
+              <ol>
+                <li>
+                  If E's style makes it an unsuitable anchor node, skip over E
+                  and its descendents.  E's style makes it an unsuitable anchor
+                  node if any of the following conditions holds:
+                  <ul>
+                    <li>E is position: fixed.
```
Comment (marcoscaceres) [inline scroll-anchoring/spec.html]: Obviously, these checks need to be defined in terms of CSSOM.

File: scroll-anchoring/spec.html [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+                    <li>E is position: absolute and E's containing block is an
+                      ancestor of the scrollable area.
+                    <li>E is position: absolute and has a top, left, right, or
+                      bottom other than 0.
+                  </ul>
+                </li>
+                <li>
+                  Otherwise, compare its bounds to the scrollable area's visible region.
+                  <ol>
+                    <li>If E is fully visible, terminate the walk and use E as the anchor node.</li>
```
Comment (marcoscaceres) [inline scroll-anchoring/spec.html]: Do we have a a formal definition of "fully visible"?

File: scroll-anchoring/spec.html [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+                    <li>If E is fully clipped (not in the viewport), skip over E and its descendents.</li>
```
Comment (marcoscaceres) [inline scroll-anchoring/spec.html]: Same with "fully clipped"

File: scroll-anchoring/spec.html [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+        Scroll anchoring aims to be the default mode of behavior when launched, so that
+        users benefit from it even on legacy content.  To allow web developers to
+        disable scroll anchoring in part or all of a webpage we propose a CSS property
+        <code>overflow-anchor</code> which applies to scrollable elements and supports the
+        following values:
+      </p>
+      <ul>
+        <li>
+            <code>overflow-anchor: visible</code> causes a scrollable area to follow the content
+            within its viewport, using the scroll anchoring algorithm described in this
```
Comment (marcoscaceres) [inline scroll-anchoring/spec.html]: Nit: link to the algo.

**Comment by marcoscaceres** [comment]
Looking good 👍

File: scroll-anchoring/spec.html (hunk 2 above)
Comment (ymalik) [inline scroll-anchoring/spec.html]: I think that's just a byproduct of copy-pasting from explainer.md. Good catch! Got rid of it.

File: scroll-anchoring/spec.html (hunk 3 above)
Comment (ymalik) [inline scroll-anchoring/spec.html]: Done.

File: scroll-anchoring/spec.html (hunk 4 above)
Comment (ymalik) [inline scroll-anchoring/spec.html]: Done.

File: scroll-anchoring/spec.html (hunk 5 above)
Comment (ymalik) [inline scroll-anchoring/spec.html]: Done.

File: scroll-anchoring/spec.html (hunk 6 above)
Comment (ymalik) [inline scroll-anchoring/spec.html]: Yes. s/we/UA everywhere.

File: scroll-anchoring/spec.html (hunk 7 above)
Comment (ymalik) [inline scroll-anchoring/spec.html]: I made unsuitable anchor a definition along with others.

File: scroll-anchoring/spec.html (hunk 8 above)
Comment (ymalik) [inline scroll-anchoring/spec.html]: Done.

File: scroll-anchoring/spec.html (hunk 9 above)
Comment (ymalik) [inline scroll-anchoring/spec.html]: By defined in terms of CSSOM, do you mean referencing to the CSSOM spec? See update.

File: scroll-anchoring/spec.html (hunk 10 above)
Comment (ymalik) [inline scroll-anchoring/spec.html]: Done.

File: scroll-anchoring/spec.html (hunk 11 above)
Comment (ymalik) [inline scroll-anchoring/spec.html]: Done.

File: scroll-anchoring/spec.html (hunk 12 above)
Comment (ymalik) [inline scroll-anchoring/spec.html]: Done.

**Comment by ymalik** [comment]
@marcoscaceres  I addressed your comments. Please take another look.

**Comment by marcoscaceres** [comment]
@ymalik thanks! LGTM.

**Comment by marcoscaceres** [comment]
@ymalik, can you please join the wicg? https://www.w3.org/community/wicg/ 

That should resolve the IPR issue being flagged above. Let me know when you do, then I can click a few magic buttons on the w3c side.

**Comment by ymalik** [comment]
@marcoscaceres I already joined the wicg. I see myself in the list of participants as well https://www.w3.org/community/wicg/participants

I am not sure why I still see the IPR issue.

**Comment by marcoscaceres** [comment]
@ymalik, thanks for letting me know. Will poke around on the W3C side and get it fixed.

**Comment by marcoscaceres** [comment]
@ymalik, ok - all good now 👍 Merge when ready
